apps/sertra/pages/server/[id].tsx

```diff
@@ -30,8 +30,34 @@ export const getStaticProps: GetStaticProps<
   // is named appropriately.
   const listingId = context.params!.id; // eslint-disable-line @typescript-eslint/no-non-null-assertion
   const res = await fetch(ApiURLs.ServerDetail(listingId));
+  const mods_res = await fetch(TsApiURLs.V1Packages("v-rising"));
   const data = await res.json();
+  const mods_data = await mods_res.json();
 
+  const updated_listing_mods_data = [];
+  for (const mod_ref of data.mods) {
+    for (const mod of mods_data) {
+      const [mod_ref_owner, mod_ref_name, mod_ref_version] = mod_ref.split(
```

Comment:
This isn't a valid way to parse package references unfortunately (I wish it was). The data includes a `full_name` property for each package version, so I'd suggest you
1. Check if the package `full_name` matches the start of the mod definition.
2. If yes, compare against all versions of the partially matching package until you find a full match.

Reply:
If you really want to implement a parser, you can copy the implementation from here: https://github.com/thunderstore-io/Thunderstore/blob/master/django/thunderstore/repository/package_reference.py but I'd recommend against it as it's unnecessary complexity in this case.

Reply:
I'd like to do 1, but the problem is that owner and the mod name are flipped. I'll figure something out that doesn't need the parser and doesn't look like hell

Reply:
Hmm, hold on they aren't flipped, dunno what I thought yesterday, nvm.

Reply:
@MythicManiac Please go over the newest implementation. I did both ways, so that you(or someone) could tell me if it's a correctish way to do the parsing?

Reply:
I'm still not sure why are you attempting to parse it instead of doing simple string comparison? If you do want to parse it this is definitely not how we want it written in code; there should be a standalone function/class handling the parsing rather than inlining it like it has been now written.

That said unless there's a reason you need to do the parsing, I'd just stick with the string comparison approach.

Reply:
You should also move the package data fetching to some standalone class/function rather than inlining it to that specific view, chances are it's going to be re-used elsewhere too and it makes sense to cache it, which isn't really possible if it's inlined the way it is currently.

That's perhaps outside the scope of this PR however, but noting it anyway for future reference.